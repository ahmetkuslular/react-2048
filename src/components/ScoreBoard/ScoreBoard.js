import React from 'react';
import styled from 'styled-components';

function ScoreBoard({ score }) {
  return (
    <Wrapper>
      <GameNameLabel>2048</GameNameLabel>
      <Scores>
        <Box>
          <Label>Score</Label>
          <Score>{score}</Score>
        </Box>
        <Box>
          <Label>Best</Label>
          <Score>10000</Score>
        </Box>
      </Scores>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const GameNameLabel = styled.div`
  font-weight: bold;
  font-size: 4em;
  text-align: center;
  color: ${props => props.theme.labelColor};
`;

const Scores = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
`;

const Box = styled.div`
  min-width: 75px;
  text-align: center;
  justify-content: center;
  background-color: ${props => props.theme.primaryColor};
  margin-left: 5px;
  padding: 5px;
  border-radius: 5px;
`;

const Label = styled.div`
  color: ${props => props.theme.labelColor};
  font-weight: bold;
  font-size: 15px;
  padding: 5px;
  text-transform: uppercase;
`;

const Score = styled.div`
  color: ${props => props.theme.white};
  font-weight: bold;
  font-size: 30px;
  left: 0;
`;

export default ScoreBoard;
