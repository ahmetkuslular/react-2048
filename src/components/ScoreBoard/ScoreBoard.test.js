
import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import ScoreBoard from './ScoreBoard';

describe('components/ScoreBoard', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ScoreBoard />, div);
  });

  it('renders markup correctly', () => {
    const tree = toJson(render(<ScoreBoard score={12} bestScore={20148}/>));
    expect(tree).toMatchSnapshot();
  });
});

