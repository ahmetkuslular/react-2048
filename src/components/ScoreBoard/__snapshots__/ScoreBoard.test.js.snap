// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ScoreBoard renders markup correctly 1`] = `
<div
  class="sc-bdVaJa UTvpt"
>
  <div
    class="sc-bwzfXH bVQOI"
  >
    2048
  </div>
  <div
    class="sc-htpNat dlPFFf"
  >
    <div
      class="sc-bxivhb cmbYYR"
    >
      <div
        class="sc-ifAKCX fWampi"
      >
        Score
      </div>
      <div
        class="sc-EHOje hnfNpt"
      >
        12
      </div>
    </div>
    <div
      class="sc-bxivhb cmbYYR"
    >
      <div
        class="sc-ifAKCX fWampi"
      >
        Best
      </div>
      <div
        class="sc-EHOje hnfNpt"
      >
        20148
      </div>
    </div>
  </div>
</div>
`;
